jsk_perception/node_scripts/fcn_depth_prediction.py

```diff
@@ -0,0 +1,520 @@
+import message_filters
+import rospy
+from sensor_msgs.msg import Image
+
+
+class FCNDepthPrediction(ConnectionBasedTransport):
+
+    def __init__(self):
+        super(self.__class__, self).__init__()
+        self.backend = rospy.get_param('~backend', 'chainer')
+        self.model_name = rospy.get_param('~model_name')
+        self.model_file = rospy.get_param('~model_file')
+        self.gpu = rospy.get_param('~gpu', -1)  # -1 is cpu mode
+        self.target_names = rospy.get_param('~target_names')
+        self.bg_label = rospy.get_param('~bg_label', 0)
+        self.proba_threshold = rospy.get_param('~proba_threshold', 0.0)
+        self.mean_bgr = np.array([104.00698793, 116.66876762, 122.67891434])
+        self._load_model()
+        self.pub_depth = self.advertise('~output', Image, queue_size=1)
+        self.pub_depth_raw = self.advertise(
+            '~output/depth_pred_raw', Image, queue_size=1)
+        self.pub_label = self.advertise('~output/label', Image, queue_size=1)
+        self.pub_proba = self.advertise(
+            '~output/proba_image', Image, queue_size=1)
+
+    def _load_model(self):
+        if self.backend == 'chainer':
+            self._load_chainer_model()
+        else:
+            raise RuntimeError('Unsupported backend: %s', self.backend)
+
+    def _load_chainer_model(self):
+        n_class = len(self.target_names)
+        if self.model_name == 'fcn8s_depth_prediction':
+            self.model = FCN8sDepthPrediction(n_class=n_class)
+        else:
+            raise ValueError(
+                'Unsupported ~model_name: {}'.format(self.model_name))
+        rospy.loginfo('Loading trained model: {0}'.format(self.model_file))
+        if self.model_file.endswith('.npz'):
+            S.load_npz(self.model_file, self.model)
+        rospy.loginfo(
+            'Finished loading trained model: {0}'.format(self.model_file))
+        if self.gpu != -1:
+            self.model.to_gpu(self.gpu)
+        if LooseVersion(chainer.__version__) < LooseVersion('2.0.0'):
+            self.model.train = False
+
+    def subscribe(self):
+        queue_size = rospy.get_param('~queue_size', 10)
+        sub_img = message_filters.Subscriber(
+            '~input', Image, queue_size=1, buff_size=2**24)
+        sub_depth = message_filters.Subscriber(
+            '~input/depth', Image, queue_size=1, buff_size=2**24)
+        self.subs = [sub_img, sub_depth]
+        if rospy.get_param('~approximate_sync', False):
+            slop = rospy.get_param('~slop', 0.1)
+            sync = message_filters.ApproximateTimeSynchronizer(
+                fs=self.subs, queue_size=queue_size, slop=slop)
+        else:
+            sync = message_filters.TimeSynchronizer(
+                fs=self.subs, queue_size=queue_size)
+        sync.registerCallback(self._cb)
+
+    def unsubscribe(self):
+        for sub in self.subs:
+            sub.unregister()
+
+    def colorize_depth(self, depth, min_value=None, max_value=None):
```

Comment:
maybe it can be a standalone function instead of a class method.

Reply:
Thanks for review!
Updated.